### PR TITLE
replaces jscs with eslint

### PR DESCRIPTION
**Pull request by PseudoNinja**
Component: Core Dependency Library
 

The JSCS project has been archived and now contains security flaws.  Migrated to ESLint to continue to provide linting service while alleviating the security concerns.
 
Fixes #15393

### CODE REVIEW

**Comment by melloware** [comment]
Nice concise PR!

**Review by melloware — APPROVED** [review]

**Comment by PseudoNinja** [comment]
Resolves bug [15393](https://bugs.jqueryui.com/ticket/15393)

File: eslint.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "no-constant-condition": 1,
+        "no-control-regex" : 0,
+        "no-empty": 1,
+        "no-mixed-spaces-and-tabs": 1,
+        "no-redeclare": 0,
+        "no-shadow-restricted-names": 1,
+        "no-unexpected-multiline": 1,
+        "no-useless-escape": 1
+    }
+}
```
Comment (mgol) [inline eslint.json:24]: Please add an empty new line at the end of the file.

File: eslint.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+    "extends": "eslint:recommended",
```
Comment (mgol) [inline eslint.json:2]: UI should use the jQuery config, similarly as jQuery itself does: https://github.com/jquery/jquery-ui/pull/1946

**Review by mgol — COMMENTED** [review]
Thanks for the PR. I think this needs some changes, though, and that may generate source changes that will be non-trivial. It's worth checking, though.

I'd expect this to remove JSHint as well.

File: eslint.json (hunk 1 above)
Comment (PseudoNinja) [reply eslint.json]: added.

File: eslint.json (hunk 2 above)
Comment (PseudoNinja) [reply eslint.json]: Im not sure we can get away from using the standardized eslint configuration file

**Comment by PseudoNinja** [comment]
Went ahead and removed jshint from the package. Let me know if there are any other changes you would like to see

File: Gruntfile.js [hunk 3]
```diff
@@ -176,25 +176,11 @@ grunt.initConfig( {
...
-		tests: {
-			options: {
-				config: true,
-				maximumLineLength: null
-			},
-			files: {
-				src: [ "tests/**/*.js" ]
-			}
-		}
+		target:["demos/**/*.js", "build/**/*.js", "ui/**/*.js"]
```
Comment (mgol) [inline Gruntfile.js:183]: This doesn't follow the jQuery code style with regards to spacing; doesn't ESLint verify it?

File: eslint.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+	"extends": "eslint:recommended",
```
Comment (mgol) [inline eslint.json:2]: jQuery projects are not supposed to use this preset but the jQuery one. This is to me a prerequisite to the migration, we can't migrate to a preset that doesn't verify rules jQuery cares about.

**Review by mgol — CHANGES_REQUESTED** [review]
The current preset is a blocker issue to me, unfortunately.

File: eslint.json (hunk 4 above)
Comment (PseudoNinja) [reply eslint.json:2]: which preset would you recommend then?

Comment (mgol) [reply eslint.json:2]: The official jQuery one: https://www.npmjs.com/package/eslint-config-jquery

**Comment by mgol** [comment]
I did the ESLint work in #1958, so I'm closing this PR.

**Pull request closed without merging**